dart/lib/vm_service_lib.dart

```diff
@@ -29,13 +29,20 @@ const String undocumented = 'undocumented';
 /// This is useful for handling the results of the Stdout or Stderr events.
 String decodeBase64(String str) => utf8.decode(base64.decode(str));
 
-Object createServiceObject(dynamic json) {
+// Returns true if a response is the Dart `null` instance.
+bool _isNullInstance(Map json) =>
+    ((json['type'] == '@Instance') && (json['kind'] == 'Null'));
+
+Object createServiceObject(dynamic json, [List<String> expectedTypes]) {
   if (json == null) return null;
 
   if (json is List) {
-    return json.map((e) => createServiceObject(e)).toList();
+    return json.map((e) => createServiceObject(e, expectedTypes)).toList();
   } else if (json is Map) {
     String type = json['type'];
+    if (_isNullInstance(json) && (!expectedTypes.contains(type))) {
```

Comment:
`expectedTypes` is an optional param; do we need it to be, or is it always passed in? I think it should probably be non-optional, but if we keep it as optional, we should check `expectedTypes` for null here

Reply:
Good call. I've made `expectedTypes` required and hardcoded `const ['Event']` as the parameter for the call sites that weren't providing that parameter.